### ISSUE TITLE
Add `enumerate` to collections

### ISSUE THREAD

**Issue by magnus-madsen** [flix/flix#4155]
See https://github.com/flix/flix/issues/3950 for details.

What type classes? Foldables?

**Comment by jaschdoc** [comment]
Yes, but eventually this should be related to `Iterable` instances (when we magically solve that issue)
